scripts/test/pre-install-airflow.sh

```diff
@@ -13,13 +17,35 @@ fi
 
 echo "${VIRTUAL_ENV}"
 
-CONSTRAINT_URL="https://raw.githubusercontent.com/apache/airflow/constraints-$AIRFLOW_VERSION.0/constraints-$PYTHON_VERSION.txt"
-curl -sSL $CONSTRAINT_URL -o /tmp/constraint.txt
+if [ "$AIRFLOW_VERSION" = "3.0" ] ; then
+  CONSTRAINT_URL="https://raw.githubusercontent.com/apache/airflow/constraints-$AIRFLOW_VERSION.2/constraints-$PYTHON_VERSION.txt"
+else
+  CONSTRAINT_URL="https://raw.githubusercontent.com/apache/airflow/constraints-$AIRFLOW_VERSION.0/constraints-$PYTHON_VERSION.txt"
+fi;
+
+curl -sSL "$CONSTRAINT_URL" -o /tmp/constraint.txt
 # Workaround to remove PyYAML constraint that will work on both Linux and MacOS
 sed '/PyYAML==/d' /tmp/constraint.txt > /tmp/constraint.txt.tmp
 mv /tmp/constraint.txt.tmp /tmp/constraint.txt
-# Install Airflow with constraints
-uv pip install "apache-airflow==$AIRFLOW_VERSION" --constraint /tmp/constraint.txt
 
-uv pip install apache-airflow-providers-cncf-kubernetes --constraint /tmp/constraint.txt
+pip install uv
+uv pip install pip --upgrade
+
+
+if [ "$AIRFLOW_VERSION" = "3.0" ]; then
+  uv pip install "apache-airflow>=3.0.2" --constraint /tmp/constraint.txt
+else
+  uv pip install "apache-airflow==$AIRFLOW_VERSION" --constraint /tmp/constraint.txt
+  uv pip install apache-airflow-providers-cncf-kubernetes --constraint /tmp/constraint.txt
+fi;
+
 rm /tmp/constraint.txt
+
+actual_version=$(airflow version | cut -d. -f1,2)
+
+if [ "$actual_version" = $AIRFLOW_VERSION ]; then
```

Comment:
Interestingly, we're having to handle this sort of logic both in Cosmos and DAG Factory. In the future, if we continue to reuse this pattern, it may make sense to create a dedicated library used by our open-source projects.